website/content/en/pre-docs/tasks/running-pods.md

```diff
@@ -0,0 +1,222 @@
+
+## Disruption budget (`PodDisruptionBudget`)
+
+Disruption budgets let you specify how much of a Deployment, ReplicationController, ReplicaSet, and StatefulSet must be protected from disruptions when pod eviction requests are made.
+This feature can be used to strike a balance between protecting the application's availability while still allowing a cluster operator to manage the cluster.
+For example:
+
+```
+apiVersion: policy/v1
+kind: PodDisruptionBudget
+metadata:
+  name: myapp-pdb
+spec:
+  minAvailable: 4
+  selector:
+    matchLabels:
+      app: myapp
+```
+In this example, the `myapp` pod is required to have at least 4 pods available after eviction.
+Instead of setting a number, you could set a percentage.
+So in this case if there were 10 pods, to get the same `minAvailable` you would set it to 40%.
+
+See [Specifying a Disruption Budget for your Application](https://kubernetes.io/docs/tasks/run-application/configure-pdb/) for details.
+
+## Node selection (`nodeSelector` and `nodeName`)
+
+With `nodeSelector` you can ask for a node that matches selected key-value pairs.
+This can include well-known labels or custom labels you create yourself.
+
+While `nodeSelector` is like node affinity, it doesn't have the same "and/or" matchExpressions that affinity has.
+So all key-value pairs must match if you use `nodeSelector`.
+Also, `nodeSelector` can do only do inclusions, while `affinity can do inclusions and exclusions (`In` and `NotIn`).
+Here is an example:
+
+```
+nodeSelector:
+  topology.kubernetes.io/zone: us-west-2a
+  node.k8.aws/capacity-type: spot
+  jones.dev: john
+```
+This example features a well-known label (`topology.kubernetes.io/zone`), a label that is specific to the AWS cloud provider (`node.k8.aws/capacity-type`), and a custom label (`jonees.dev`).
+Use a command like the following to set a custom label:
+
+```
+kubectl label nodes <node_name> <label_key>=<label_value>
+```
+
+See [nodeSelector](https://kubernetes.io/docs/concepts/scheduling-eviction/assign-pod-node/#nodeselector) in the Kubernetes documentation for details.
+
+To select a node by name, use the `nodeName` object:
+
+```
+apiVersion: v1
+kind: Pod
+metadata:
+  name: myapp
+spec:
+  containers:
+  - name: myapp
+    image: myapp
+  nodeName: my-node
+```
+## Topology spread (`topologySpreadConstraints`)
+
+By using the Kubernetes `topologySpreadConstraints` you can ask the provisioner to have pods push away from each other for high availability reasons.
+Think of it as the Kubernetes evolution for pod affinity: it lets you relate pods with respect to nodes while still allowing spread.
+For example:
+
+```
+topologySpreadConstraints:
+  - maxSkew: 1
+    topologyKey: "topology.kubernetes.io/zone"
+    topologyKey: "topology.kubernetes.io/hostname"
```

Comment:
This would need to be two different rules. This is currently invalid YAML (topologyKey is defined twice). IIUC, you also need a selector for a valid rule.

Reply:
Okay. I did this:
`spec:
  topologySpreadConstraints:
    - maxSkew: 1
      topologyKey: "topology.kubernetes.io/zone"
      whenUnsatisfiable: ScheduleAnyway
      labelSelector:
        matchLabels:
          dev: jjones
  topologySpreadConstraints:
    - maxSkew: 1
      topologyKey: "topology.kubernetes.io/node"
      whenUnsatisfiable: ScheduleAnyway
      labelSelector:
        matchLabels:
          dev: jjones
`